internal/wasi/fs_test.go

```diff
@@ -12,10 +17,242 @@ import (
 func TestMemFile_Read_Seek(t *testing.T) {
 	expectedFileContent := []byte("wazero") // arbitrary contents
 	memFile := &memFile{
-		buf: expectedFileContent,
+		fsEntry: &memFSEntry{Contents: expectedFileContent},
 	}
 	// TestReader tests that io.Reader correctly reads the expected contents.
 	// It also tests io.Seeker when it's implemented, which memFile does.
 	err := iotest.TestReader(memFile, expectedFileContent)
```

Comment:
maybe mention why we aren't using more of the iotest, yet, at the top of this file.. ex an overview comment about use of iotest and fstest (including status) then the functions